helm/charts/oathkeeper/values.yaml

```diff
@@ -88,26 +88,30 @@ oathkeeper:
 #  If set, uses the given access rules.
 #  accessRules: {}
 
-resources: {}
-#  We usually recommend not to specify default resources and to leave this as a conscious
-#  choice for the user. This also increases chances charts run on environments with little
-#  resources, such as Minikube. If you do want to specify resources, uncomment the following
-#  lines, adjust them as necessary, and remove the curly braces after 'resources:'.
-#  limits:
-#    cpu: 100m
-#    memory: 128Mi
-#  requests:
-#    cpu: 100m
-#  memory: 128Mi
+deployment: 
+  resources: {}
+  #  We usually recommend not to specify default resources and to leave this as a conscious
+  #  choice for the user. This also increases chances charts run on environments with little
+  #  resources, such as Minikube. If you do want to specify resources, uncomment the following
+  #  lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  #  limits:
+  #    cpu: 100m
+  #    memory: 128Mi
+  #  requests:
+  #    cpu: 100m
+  #  memory: 128Mi
 
-# Node labels for pod assignment.
-nodeSelector: {}
-# If you do want to specify node labels, uncomment the following
-# lines, adjust them as necessary, and remove the curly braces after 'annotations:'.
-#   foo: bar
+  # Node labels for pod assignment.
+  nodeSelector: {}
+  # If you do want to specify node labels, uncomment the following
+  # lines, adjust them as necessary, and remove the curly braces after 'annotations:'.
+  #   foo: bar
 
-# Configure node tolerations.
-tolerations: []
+  # Configure node tolerations.
+  tolerations: []
 
-# Configure node affinity
-affinity: {}
+  # Configure node affinity
+  affinity: {}
+
+  # Annotations
+  annotations: {}
```

Comment:
I think it would be better if we just have e.g. `deployment.annotations` or `deploymentAnnotations`. The issue with this change is that backwards compatibility for these keys will be broken. Since `resources`, `nodeSelector`, `tolerations` and `affinity` are all used only in deployment I think it is ok to have them as top level key. For annotations, this is not the case however.

What do you think?

Reply:
In the original chart of oathkeeper, there is no annotation field in deployment or services, so we don't have any backward compatibility issues here. Yes I agree that you can have annotations or services or ingresses, but a good practice is to share them across all items in a chart. 
In the case of oathkeeper and maester, i just don't see much use in annotation in objects other than the deployments